src/test/txreconciliation_tests.cpp

```diff
@@ -0,0 +1,673 @@
+
+    NodeId peer_id0 = 0;
+    bool inbound = true;
+    tracker.SuggestReconciling(peer_id0, inbound);
+    tracker.EnableReconciliationSupport(peer_id0, inbound, true, false, 1, 0, 0);
+    assert(tracker.IsPeerRegistered(peer_id0));
+
+    assert(tracker.GetPeerSetSize(peer_id0) == 0);
+
+    size_t count = 10;
+    for (size_t i = 0; i < count; i++) {
+        tracker.AddToReconSet(peer_id0, std::vector<uint256>{GetRandHash()});
+    }
+
+    assert(tracker.GetPeerSetSize(peer_id0) == count);
+}
+
+BOOST_AUTO_TEST_CASE(MaybeRequestReconciliationTest)
+{
+    TxReconciliationTracker tracker;
+    NodeId peer_id0 = 0;
+    int64_t start_time = GetTime();
+    SetMockTime(start_time);
+
+    // Don't request from a non-registered peer.
+    for (int i = 0; i < 10; i++) {
+        SetMockTime(start_time + i * 10);
+        assert(!tracker.MaybeRequestReconciliation(peer_id0));
+    }
+
+    tracker.SuggestReconciling(peer_id0, true);
+    tracker.EnableReconciliationSupport(peer_id0, true, true, false, 1, 0, 0);
+
+    SetMockTime(start_time);
+    // Don't request from an inbound peer.
+    for (int i = 0; i < 10; i++) {
+        SetMockTime(start_time + i * 10);
+        assert(!tracker.MaybeRequestReconciliation(peer_id0));
+    }
+
+    NodeId peer_id1 = 1;
+    tracker.SuggestReconciling(peer_id1, false);
+    tracker.EnableReconciliationSupport(peer_id1, false, false, true, 1, 0, 0);
+
+    // Make a request.
+    size_t requested = 0;
+    SetMockTime(start_time);
+    for (int i = 0; i < 10; i++) {
+        SetMockTime(start_time + i * 10);
+        auto request_data = tracker.MaybeRequestReconciliation(peer_id1 );
+        if (request_data) {
+            const auto [local_set_size, local_q_formatted] = (*request_data);
+            assert(local_set_size == 0);
+            assert(local_q_formatted == uint16_t(DEFAULT_RECON_Q * Q_PRECISION));
+            ++requested;
+        }
+    }
+    // Request only once until it's responded.
+    assert(requested == 1);
+
+    // If the reconciliation is over, the next request should happen after some interval.
+    NodeId peer_id2 = 2;
+    tracker.SuggestReconciling(peer_id2, false);
+    tracker.EnableReconciliationSupport(peer_id2, false, false, true, 1, 0, 0);
+    SetMockTime(start_time + 10000);
+    assert(tracker.MaybeRequestReconciliation(peer_id2));
+    std::vector<uint32_t> txs_to_request;
+    std::vector<uint256> txs_to_announce;
+    std::optional<bool> recon_result;
+    tracker.HandleSketch(peer_id2, std::vector<uint8_t>(), txs_to_request, txs_to_announce, recon_result);
+    assert(recon_result && !*recon_result); // Check it's finalized.
+
+    assert(!tracker.MaybeRequestReconciliation(peer_id2));
+    SetMockTime(start_time + 20000);
+    // Still don't request because the other peer is earlier in the queue.
+    assert(!tracker.MaybeRequestReconciliation(peer_id2));
+    tracker.RemovePeer(peer_id1);
+    assert(tracker.MaybeRequestReconciliation(peer_id2));
+}
```

Comment:
Maybe add some transactions to the set, to exercise that the correct `local_set_size` is returned 

```
@@ -264,6 +270,25 @@ BOOST_AUTO_TEST_CASE(MaybeRequestReconciliationTest)
     assert(!tracker.MaybeRequestReconciliation(peer_id2));
     tracker.RemovePeer(peer_id1);
     assert(tracker.MaybeRequestReconciliation(peer_id2));
+    tracker.RemovePeer(peer_id2);
+
+    NodeId peer_id3 = 3;
+    tracker.SuggestReconciling(peer_id3, false);
+    tracker.EnableReconciliationSupport(peer_id3, false, false, true, 1, 0, 0);
+
+    size_t count = 10;
+    for (size_t i = 0; i < count; i++) {
+        tracker.AddToReconSet(peer_id0, std::vector<uint256>{GetRandHash()});
+    }
+
+    SetMockTime(start_time + 20000);
+    // Make a request.
+    auto request_data = tracker.MaybeRequestReconciliation(peer_id3);
+    if (request_data) {
+        const auto [local_set_size, local_q_formatted] = (*request_data);
+        assert(local_set_size == 10);
+        assert(local_q_formatted == uint16_t(DEFAULT_RECON_Q * Q_PRECISION));
+    }
 }
```